src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -394,9 +420,39 @@ protected override void OnHostStarted()
                 _secretManager.PurgeOldSecretsAsync(Instance.ScriptConfig.RootScriptPath, Instance.TraceWriter, Instance.Logger);
             }
 
+            if (Microsoft.Azure.WebJobs.Script.WebHost.Management.FunctionsSyncManager.IsSyncTriggersEnvironment(_settingsManager))
+            {
+                // create a onetime invocation timer
+                if (_syncTimer != null)
+                {
+                    // cancel any outstanding timer
+                    _syncTimer.Change(Timeout.Infinite, Timeout.Infinite);
+                }
+                _syncTimer = new Timer(OnSyncTimerTick, CancellationToken.None, 30 * 1000, Timeout.Infinite);
```

Comment:
Are you always passing `CancellationToken.None` here? I see you check it in `OnSyncTimerTick`, but it'll never be canceled, will it?

Reply:
Good catch - changed to null